datasets/clinc_oos/clinc_oos.py

```diff
@@ -0,0 +1,272 @@
+
+import json
+import textwrap
+
+import datasets
+
+
+_CITATION = """\
+    @inproceedings{larson-etal-2019-evaluation,
+    title = "An Evaluation Dataset for Intent Classification and Out-of-Scope Prediction",
+    author = "Larson, Stefan  and
+      Mahendran, Anish  and
+      Peper, Joseph J.  and
+      Clarke, Christopher  and
+      Lee, Andrew  and
+      Hill, Parker  and
+      Kummerfeld, Jonathan K.  and
+      Leach, Kevin  and
+      Laurenzano, Michael A.  and
+      Tang, Lingjia  and
+      Mars, Jason",
+    booktitle = "Proceedings of the 2019 Conference on Empirical Methods in Natural Language Processing and the 9th International Joint Conference on Natural Language Processing (EMNLP-IJCNLP)",
+    year = "2019",
+    url = "https://www.aclweb.org/anthology/D19-1131"
+}
+"""
+
+_DESCRIPTION = """\
+    This dataset is for evaluating the performance of intent classification systems in the
+    presence of "out-of-scope" queries. By "out-of-scope", we mean queries that do not fall
+    into any of the system-supported intent classes. Most datasets include only data that is
+    "in-scope". Our dataset includes both in-scope and out-of-scope data. You might also know
+    the term "out-of-scope" by other terms, including "out-of-domain" or "out-of-distribution".
+"""
+
+_DESCRIPTIONS = {
+    "small": textwrap.dedent(
+        """\
+        Small, in which there are only 50 training queries per each in-scope intent
+        """
+    ),
+    "imbalanced": textwrap.dedent(
+        """\
+        Imbalanced, in which intents have either 25, 50, 75, or 100 training queries.
+        """
+    ),
+    "plus": textwrap.dedent(
+        """\
+        OOS+, in which there are 250 out-of-scope training examples, rather than 100.
+        """
+    ),
+}
+
+_URL = "https://github.com/clinc/oos-eval/"
+
+_DATA_URLS = {
+    "small": "https://raw.githubusercontent.com/clinc/oos-eval/master/data/data_small.json",
+    "imbalanced": "https://raw.githubusercontent.com/clinc/oos-eval/master/data/data_imbalanced.json",
+    "plus": "https://raw.githubusercontent.com/clinc/oos-eval/master/data/data_oos_plus.json",
+}
+
+
+class ClincConfig(datasets.BuilderConfig):
+
+    """BuilderConfig for CLINIC150"""
+
+    def __init__(self, description, data_url, citation, url, **kwrags):
+        """
+        Args:
+            description: `string`, brief description of the dataset
+            data_url: `dictionary`, dict with url for each split of data.
+            citation: `string`, citation for the dataset.
+            url: `string`, url for information about the dataset.
+            **kwrags: keyword arguments frowarded to super
+        """
+        super(ClincConfig, self).__init__(version=datasets.Version("1.0.0", ""), **kwrags)
+        self.description = description
+        self.data_url = data_url
+        self.citation = citation
+        self.url = url
+
+
+class Clinc150(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class ClincOos(datasets.GeneratorBasedBuilder):
```
this needs to be the camle case version of the dataset script name `clinc_oos`